build_tools/cmake/iree_static_linker_test.cmake

```diff
@@ -0,0 +1,259 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+include(CMakeParseArguments)
+
+# iree_static_linker_test()
+#
+# Creates a test with statically linked libraries. Assuming the target backend
+# is llvm-cpu and the driver is the local-sync driver.
+#
+# Parameters:
+#   NAME: Name of the target
+#   SRC: mlir source file to be compiled to an IREE module.
+#   INPUT_TYPE: Module input type, assuming native c/c++ data types
+#   HAL_TYPE: HAL data type, see runtime/src/iree/hal/buffer_view.h for possible
+#       options.
+#   STATIC_LIB_PREFIX: llvm static library prefix.
+#   MAIN_FUNCTION: Entry function call.
+#   EMITC: Optional, use emitC instead of vmfb module to define the vm contrl.
```

Comment:
nit: "Options" arguments (keywords that don't use any values, "if set, do X") can use slightly different wording, like these: https://github.com/iree-org/iree/blob/e33c64c8c1b58958cf56982372e53332c5c1aeb2/build_tools/cmake/iree_c_embed_data.cmake#L20-L21

```suggestion
#   EMITC: Uses EmitC to output C code instead of VM bytecode.
```

Reply:
Fixed.